index.d.ts

```diff
@@ -0,0 +1,320 @@
+declare module 'fluture' {
+
+  interface RejectFunction<L> {
+    (error: L): void
+  }
+
+  interface ResolveFunction<R> {
+    (value: R): void
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  interface Nodeback<E, R> {
+    (err: E | null, value?: R): void
+  }
+
+  interface Next<T> {
+    done: false
+    value: T
+  }
+
+  interface Done<T> {
+    done: true
+    value: T
+  }
+
+  interface Iterator<N, D> {
+    next(value?: N): Next<N> | Done<D>
+  }
+
+  interface Generator<Y, R> {
+    (): Iterator<Y, R>
+  }
```

Comment:
I don't know. By the looks of that interface, `Next<A>` and `Done<B>` must be expressed as one `IteratorResult<A|B>`. I don't like that because it means we would have to relax the type signature for `chainRec`.